server/schedule/filter/filters.go

```diff
@@ -264,135 +264,206 @@ func (f *distinctScoreFilter) GetSourceStoreID() uint64 {
 	return f.srcStore
 }
 
-// StoreStateFilter is used to determine whether a store can be selected as the
-// source or target of the schedule based on the store's state.
-type StoreStateFilter struct {
+// The condition table.
+// Y: the condition is temporary (expected to become false soon).
+// N: the condition is expected to be true for a long time.
+// X means when the condition is true, the store CANNOT be selected.
+//
+// Condition    Down Offline Tomb Pause Disconn Busy RmLimit AddLimit Snap Pending Reject
+// IsTemporary  N    N       N    N     Y       Y    Y       Y        Y    Y       N
+//
+// LeaderSource X            X    X     X
+// RegionSource                                 X    X                X
+// LeaderTarget X    X       X    X     X       X                                  X
+// RegionTarget X    X       X          X       X            X        X    X
+
+const (
+	leaderSource = iota
+	regionSource
+	leaderTarget
+	regionTarget
+	scatterRegionTarget
+)
+
+// conditionFunc defines condition to determine a store should be selected.
+// It should consider if the filter allows temporary states.
+type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
+
+// TemporaryStateFilter is used to determine whether a store can be selected as the
+// source or target of the schedule based on the store's temporary state.
+type TemporaryStateFilter struct {
 	ActionScope string
 	// Set true if the schedule involves any transfer leader operation.
 	TransferLeader bool
 	// Set true if the schedule involves any move region operation.
 	MoveRegion bool
 	// Set true if the scatter move the region
 	ScatterRegion bool
```

Comment:
Could we merge the usage info `TransferLeader`,`MoveRegion`,`ScatterRegion` into one int value, such as `OpKind` did, so we could support more in the future.